yt/loaders.py

```diff
@@ -1385,10 +1386,19 @@ def load_sample(
     # however we don't want to do it right now because the "filename" column is
     # currently incomplete
 
-    try:
-        specs = registry_table.query(f"`filename` == '{topdir}'").iloc[0]
-    except IndexError as err:
-        raise KeyError(f"Could not find '{fn}' in the registry.") from err
+    known_names = registry_table.dropna().filename.to_list()
+    if topdir not in known_names:
+        msg = f"'{topdir}' is not an available dataset."
+        lexical_distances = [
+            (name, levenshtein_distance(name, topdir)) for name in known_names
+        ]
+        suggestions = [name for name, dist in lexical_distances if dist < 4]
+        if suggestions:
+            msg += " Did you mean to type any of the following ?\n\n    "
+            msg += "\n    ".join(str(_) for _ in suggestions)
+        raise ValueError(msg)
```

Comment:
to anyone wondering, I switched from KeyError to ValueError because of a known bug in Python itself https://bugs.python.org/issue2651
in short: KeyError doesn't support formatted strings